### PR TITLE
Limit `user.touch` usage in RepoSyncWorker

### PR DESCRIPTION
**Pull request by maestromac**
## What type of PR is this? (check all applicable)
- [x] Optimization

## Description
In our background worker class  `GithubRepos::RepoSyncWorker`, we are updating user's `github_repos_updated_at` after updating their repo info. There is a high likelihood that will happen multiple times because users tend to have multiple repos that are being updated consecutively. There's no need to update it every single time because it redundantly calling callbacks, slightly polluting the sidekiq queues.

## Related Tickets & Documents
n/a

## QA Instructions, Screenshots, Recordings
I believe the test is suffice
## Added tests?
- [x] yes

## Added to documentation?
- [x] no documentation needed

## [optional] Are there any post deployment tasks we need to perform?
n/a

### CODE REVIEW

File: app/workers/github_repos/repo_sync_worker.rb [hunk 1]
```diff
@@ -26,7 +26,9 @@ def perform(repo_id)
           stargazers_count: fetched_repo.stargazers_count,
           info_hash: fetched_repo.to_hash,
         )
-        repo.user&.touch(:github_repos_updated_at)
+        if repo.user && repo.user.github_repos_updated_at < 30.minutes.ago
```
Comment (citizen428) [inline app/workers/github_repos/repo_sync_worker.rb:29]: **suggestion (non-blocking):** I don't know about others, but I find the comparison operators always a bit hard when it comes to dates/times in Rails, so I use the more explicit versions instead:

```suggestion
        if repo.user && repo.user.github_repos_updated_at.before?(30.minutes.ago)
```

Comment (maestromac) [reply app/workers/github_repos/repo_sync_worker.rb:29]: Thanks @citizen428 ! In the back of my mind, I knew something like this would exist in Ruby.

**Review by rhymes — APPROVED** [review]
LGTM

File: app/workers/github_repos/repo_sync_worker.rb [hunk 2]
```diff
@@ -26,7 +26,9 @@ def perform(repo_id)
           stargazers_count: fetched_repo.stargazers_count,
           info_hash: fetched_repo.to_hash,
         )
-        repo.user&.touch(:github_repos_updated_at)
+        if repo.user&.github_repos_updated_at&.before?(30.minutes.ago)
```
Comment (mstruve) [inline app/workers/github_repos/repo_sync_worker.rb:29]: Should we make 30.minutes a constant here?

**Review by mstruve — APPROVED** [review]
Oooof thats a lot of user updating! Good catch

File: app/workers/github_repos/repo_sync_worker.rb (hunk 2 above)
Comment (maestromac) [reply app/workers/github_repos/repo_sync_worker.rb:29]: Updated!

**Pull request merged**